packages/askui-nodejs/src/execution/ui-control-client.ts

```diff
@@ -588,26 +610,247 @@ export class UiControlClient extends ApiCommands {
+   *
+   * **Examples:**
+   * ```typescript
+   * await aui.expectExistence([
+   *   {
+   *     type: 'text',
+   *     label: 'Switch to Dark',
+   *     matching: 'similar'
+   *   },
+   * ]);
+   *
+   * // Check for existence of multiple elements
+   * await aui.expectExistence([
+   *   {
+   *     type: 'textfield',
+   *     label: '42',
+   *     relation: {
+   *       type: 'below',
+   *       text: 'Register'
+   *     }
+   *   },
+   *   {
+   *     type: 'element',
+   *     label: 'Switch to Dark',
+   *   },
+   * ]);
+   *
+   * // Validate existence
+   * const exists = await aui.expectExistence([...]);
+   * exists.everythingExists // true when every element exists
+   *
+   * // Check which elements do not exist
+   * // with the elements property
+   * const nonExistentElements = exists.elements.filter((e) => e.exists===false)
+   * ```
+   *
+   * @param {ExpectExistenceInputParameter[]} params - Objects containing the required properties
+   *                                                   'type', 'label', 'matching' and 'relation'.
+   * @property {string} params.type - The type of the element: 'text' | 'button' | 'checkbox' |
+   *                                                           'switch' | 'textfield' | 'element'
+   * @property {string} params.label - The label text for the element.
+   *                                   In case of 'text': The text to be matched for
+   * @property {string} params.matching - Whether the text is matched using similarity,
+   *                                      exact match or a regular expression.
+   * @property {Object} params.relation - Object describing the relationship between the
+   *                                      clicked text and another element.
+   * @property {RelationsForConvenienceMethods} params.relation.type - The type of relation.
+   * @property {string} params.relation.text - The label or text associated with the
+   *                                           related element or state.
+   * @returns {ExpectExistenceReturnValue} - Property boolean 'everythingExists' tells
+   *                                         if every element exists.
+   */
+  async expectExistence(
+    params: ExpectExistenceInputParameter[],
+  ): Promise<ExpectExistenceReturnValue> {
+    if (params === undefined) {
+      throw new ValidationError('At least one ExpectExistenceInputParameter has to be there.');
+    }
+
+    const returnValue: ExpectExistenceReturnValue = {
+      everythingExists: true,
+      elements: [],
+    };
+
+    for (let i = 0; i < params.length; i += 1) {
+      const element = params[i] as ExpectExistenceInputParameter;
+
+      switch (element.type) {
+        case 'text': {
+          let command = this.get().text();
+          command = await this.evaluateMatchingProperty(
+            command,
+            element.matching,
+            element.label,
+          ) as FluentFiltersOrRelationsGetter;
```

Comment:
If you really need a type cast, I would put it into the method and use the correct return type. But ideally, I would try to omit it by fixing the typing.